File: Java/src/main/java/Solution901_1000/Solution999.java
package Solution901_1000;

/**
 * @author windows
 */
public class Solution999 {
    public int numRookCaptures(char[][] board) {
        int[] pos = new int[2];
        for (int i = 0; i < board.length; i++) {
            for (int j = 0; j < board[i].length; j++) {
                if (board[i][j] == 'R'){
                    pos[0] = i;
                    pos[1] = j;
                    break;
                }
            }
        }
        int count = 0;
        for (int i = pos[0]; i >= 0; i--) {
            if (board[i][pos[1]] == 'B'){
                break;
            }else if(board[i][pos[1]] == 'p'){
                count++;
                break;
            }
        }
        for (int i = pos[0]; i < board.length; i++) {
            if (board[i][pos[1]] == 'B'){
                break;
            }else if(board[i][pos[1]] == 'p'){
                count++;
                break;
            }
        }
        for (int i = pos[1]; i >= 0; i--) {
            if (board[pos[0]][i] == 'B'){
                break;
            }else if(board[pos[0]][i] == 'p'){
                count++;
                break;
            }
        }
        for (int i = pos[1]; i < board[0].length; i++) {
            if (board[pos[0]][i] == 'B'){
                break;
            }else if(board[pos[0]][i] == 'p'){
                count++;
                break;
            }
        }
        return count;
    }


    public static void main(String[] args) {
        Solution999 solution999 = new Solution999();
    }
}

File: C++/src/Solution901_1000/Solution999.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution999.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;


int numRookCaptures(vector<vector<char>>& board) {
  int m = board.size();
  int n = board[0].size();
  int count = 0;
  for (int i = 0; i < m; ++i) {
    for (int j = 0; j < n; ++j) {
      if (board[i][j] == 'R') {
        for (int k = i; k >= 0; --k) {
          if (board[k][j] == 'p') {
            count++;
            break;
          }else if (board[k][j] == 'B'){
            break;
          }
        }

        for (int k = i + 1; k < m; ++k) {
          if (board[k][j] == 'p') {
            count++;
            break;
          }else if (board[k][j] == 'B'){
            break;
          }
        }

        for (int k = j; k >= 0; --k) {
          if (board[i][k] == 'p') {
            count++;
            break;
          }else if (board[i][k] == 'B'){
            break;
          }
        }

        for (int k = j + 1; k < n; ++k) {
          if (board[i][k] == 'p') {
            count++;
            break;
          }else if (board[i][k] == 'B'){
            break;
          }
        }
      }
    }
  }
  return count;
}
